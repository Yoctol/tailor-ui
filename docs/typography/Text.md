---
id: text
title: Text
---

## Examples

```js
import { Text } from 'tailor-ui';
```

### Basic

```jsx live
<Box width="100%" style={{ whiteSpace: 'pre-line' }}>
  <Text
    color="primaryDark"
    letterSpacing={1.5}
    lineHeight="2"
    textAlign="center"
  >
    Lorem ipsum dolor, sit amet consectetur adipisicing elit. Itaque ullam
    perspiciatis accusantium atque veniam aut quibusdam quas velit laboriosam.
    Dolorem, modi sit? Nisi dolorum hic aperiam blanditiis, modi vitae repellat?
  </Text>
</Box>
```

## API
