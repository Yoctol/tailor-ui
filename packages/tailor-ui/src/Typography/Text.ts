import styled from 'styled-components';
import {
  ColorProps,
  TypographyProps,
  color,
  compose,
  typography,
} from 'styled-system';

export type TextProps = ColorProps & TypographyProps;

export const text = compose(color, typography);

const Text = styled('span')<TextProps>(text);

export { Text };
