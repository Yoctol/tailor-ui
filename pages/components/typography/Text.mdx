---
name: Text
menu: Typography
route: /typography/text
---

import { PropsTable, Playground } from 'docz';
import { Text } from 'tailor-ui';

# Text

## Examples

```js
import { Text } from 'tailor-ui';
```

### Basic

<Playground>
  <Text
    color="primaryDark"
    letterSpacing={1.5}
    lineHeight="2"
    textAlign="center"
  >
    Lorem ipsum dolor, sit amet consectetur adipisicing elit. Itaque ullam
    perspiciatis accusantium atque veniam aut quibusdam quas velit laboriosam.
    Dolorem, modi sit? Nisi dolorum hic aperiam blanditiis, modi vitae repellat?
  </Text>
</Playground>

## API

<PropsTable of={Text} />
