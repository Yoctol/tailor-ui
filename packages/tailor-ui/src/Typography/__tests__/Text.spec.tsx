import React from 'react';

import { render } from 'test/test-utils';

import { Text } from '../Text';

describe('Text', () => {
  it('should render correctly', () => {
    const { container } = render(
      <Text
        color="primaryDark"
        letterSpacing={1.5}
        lineHeight="2"
        textAlign="center"
      >
        Some Text
      </Text>
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
