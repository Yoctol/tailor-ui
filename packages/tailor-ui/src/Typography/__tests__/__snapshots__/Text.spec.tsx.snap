// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text should render correctly 1`] = `
.c0 {
  color: #184480;
  -webkit-letter-spacing: 1.5px;
  -moz-letter-spacing: 1.5px;
  -ms-letter-spacing: 1.5px;
  letter-spacing: 1.5px;
  line-height: 2;
  text-align: center;
  font-size: 1.125rem;
}

<span
  class="c0"
  color="primaryDark"
  font-size="lg"
  letter-spacing="1.5"
>
  Some Text
</span>
`;
